osquery/table/airdrop_darwin.go

```diff
@@ -41,14 +41,15 @@ func (t *airdropTable) generateAirdrop(ctx context.Context, queryContext table.Q
 		username = t.primaryUser
 	}
 
-	discover := "Unknown"
-	if val, ok := copyPreferenceValue("DiscoverableMode", "com.apple.sharingd", username).(string); ok {
-		discover = val
+	out, err := execPreferenceAsUser(ctx, username, "DiscoverableMode", "com.apple.sharingd")
```

Comment:
`username = q.Constraints[0].Expression` this isn't going to work if you, for example, join this table against the users table with more than 1 user.

Reply:
That is true. I don't know if joining against multiple users is ever desirable here, and would end up in a large number of sequential exec calls.

I don't know if it's the right decision, but I thought this query should only work for a single user. The default is to query the primary user anyway.